### ISSUE TITLE
New rule: dont_create_a_return_var

### ISSUE THREAD

**Issue by danylo-safonov-solid** [solid-software/solid_lints#92]
BAD:
```dart
void x() {
  final y = 1;

  return y;
}
```

GOOD:
```dart
void x() {
  return 1;
}
```

**Comment by tozpeak** [comment]
During conversation with @solid-dmytrotkachuk the following criteria was decided for now:
- it is Bad, if no statements are between declaration and return,
- it is Bad, if declaration initializer is an immutable expression (all parameters are const or final),
- it is Good, if variable is used more than in return statement itself,
- it is Good, if declaration initializer contains mutable arguments, since it would be hard to predict if this wasn't used for valid caching.
